test/workbox-strategies/sw/test-NetworkFirst.mjs

```diff
@@ -141,12 +141,37 @@ describe(`NetworkFirst`, function() {
       const cache = await caches.open(cacheNames.getRuntimeName());
       await cache.put(request, injectedResponse.clone());
 
-      const handlePromise = networkFirst.handle({
+      const [handlePromise, donePromise] = networkFirst.handleAll({
+        request,
+        event,
+      });
+
+      await eventDoneWaiting(event);
+      await donePromise;
```

Comment:
I don't you think you need both `eventDoneWaiting()` and `donePromise` (I expect them to resolve at the same time. Were you seeing a situations where that was not the case?

Reply:
In the scenario I was seeing, if the network request succeeded, then it canceled the timeout, and so donePromise never resolved. However, the handlePromise still resolved, so the bug (the unresolved promise) went unnoticed until I tried to upgrade the service worker; at that point, the promise kept the old service worker in the "running" state.

For this scenario ("should return the cached response if the network request times out"), both promises resolved even before my changes. But, since the "successful completion before timeout" scenario involved a hard-to-notice bug with only one promise remaining, it seemed worthwhile to add an explicit test for the "timeout" scenario, too.

I'm happy to delete this if you don't think it's worth keeping.

Reply:
The logic for `donePromise` is virtually identical (actually it was copy/pasted from) the logic for `eventDoneWaiting()`, so you shouldn't need both. I'd prefer to just use `donePromise` since it uses library code rather than a test helper.